src/main/java/io/jenkins/plugins/artifact_manager_jclouds/s3/S3BlobStoreConfig.java

```diff
@@ -274,51 +278,60 @@ public static S3BlobStoreConfig get() {
     }
 
     @VisibleForTesting
-    static Supplier<AmazonS3ClientBuilder> clientBuilder = AmazonS3ClientBuilder::standard;
+    static Supplier<S3ClientBuilder> clientBuilder = S3Client::builder;
 
     /**
     *
     * @return an AmazonS3ClientBuilder using the region or not, it depends if a region is configured or not.
     */
-    AmazonS3ClientBuilder getAmazonS3ClientBuilder() {
-        AmazonS3ClientBuilder ret = clientBuilder.get();
+    S3ClientBuilder getAmazonS3ClientBuilder() throws URISyntaxException {
+        S3ClientBuilder ret = clientBuilder.get();
 
-        if (StringUtils.isNotBlank(customEndpoint)) {
+        if (StringUtils.isNotBlank(getResolvedCustomEndpoint())) {
             String resolvedCustomSigningRegion = customSigningRegion;
             if (StringUtils.isBlank(resolvedCustomSigningRegion)) {
                 resolvedCustomSigningRegion = "us-east-1";
             }
-            ret = ret.withEndpointConfiguration(new AwsClientBuilder.EndpointConfiguration(getResolvedCustomEndpoint(), resolvedCustomSigningRegion));
+            ret = ret.endpointOverride(new URI(getResolvedCustomEndpoint())).region(Region.of(resolvedCustomSigningRegion));
         } else if (StringUtils.isNotBlank(CredentialsAwsGlobalConfiguration.get().getRegion())) {
-            ret = ret.withRegion(CredentialsAwsGlobalConfiguration.get().getRegion());
+            ret = ret.region(Region.of(CredentialsAwsGlobalConfiguration.get().getRegion()));
         } else {
-            ret = ret.withForceGlobalBucketAccessEnabled(true);
+            ret = ret.useArnRegion(true);
         }
-        ret = ret.withAccelerateModeEnabled(useTransferAcceleration);
+        ret = ret.accelerate(useTransferAcceleration);
 
         // TODO the client would automatically use path-style URLs under certain conditions; is it really necessary to override?
-        ret = ret.withPathStyleAccessEnabled(getUsePathStyleUrl());
+        ret = ret.forcePathStyle(getUsePathStyleUrl());
 
         return ret;
     }
 
     @VisibleForTesting
-    public AmazonS3ClientBuilder getAmazonS3ClientBuilderWithCredentials() throws IOException {
-        return getAmazonS3ClientBuilderWithCredentials(getDisableSessionToken());
+    public S3ClientBuilder getAmazonS3ClientBuilderWithCredentials() throws IOException {
```

Comment:
I'm a bit confused to see 2 places where credentials are resolved here and in S3BlobStore (looks to be only used for JCloudsArtifactManagerTest which is not using Jenkins credentials or I misunderstand something)